### PR TITLE
New op add analyze main

### PR DESCRIPTION
**Pull request by qingxinhome**
## What type of PR is this?

- [ ] API-change
- [ ] BUG
- [ ] Improvement
- [ ] Documentation
- [x] Feature
- [ ] Test and CI
- [ ] Code Refactoring

## Which issue(s) this PR fixes:

issue https://github.com/matrixorigin/matrixone/issues/10862
issue https://github.com/matrixorigin/matrixone/issues/19728

## What this PR does / why we need it:
1. Needs to estimate S3 count while do insert

### CODE REVIEW

**Review by aunjgr — APPROVED** [review]

**Review by badboynt1 — APPROVED** [review]

**Review by ouyuanning — APPROVED** [review]

File: pkg/util/trace/impl/motrace/statistic/stats_array.go [hunk 1]
```diff
@@ -183,6 +183,11 @@ func (s *StatsArray) WithS3IODeleteCount(v float64) *StatsArray {
 	return s
 }
 
+func (s *StatsArray) WithS3InputEstimateCount(v float64) *StatsArray {
```
Comment (xzxiong) [inline pkg/util/trace/impl/motrace/statistic/stats_array.go:186]: @qingxinhome 请去掉该接口，直接使用  `WithS3IOInputCount`

File: pkg/util/trace/impl/motrace/statistic/stats_array.go [hunk 2]
```diff
@@ -661,3 +666,11 @@ func EnsureStatsInfoCanBeFound(ctx context.Context, from context.Context) contex
 	ctx = context.WithValue(ctx, statsInfoKey{}, v)
 	return ctx
 }
+
+// Each 8192 rows of data corresponds to an S3 input unit
+const rowsPerS3Input int64 = 8192
+
+// Estimating S3input using the number of written lines, Maintained by @LeftHandCold
+func EstimateS3Input(writtenRows int64) float64 {
```
Comment (xzxiong) [inline pkg/util/trace/impl/motrace/statistic/stats_array.go:674]: 这个函数不应该放在这里,  应放置在 @LeftHandCold 维护的 pkg

File: pkg/sql/compile/analyze_module.go [hunk 3]
```diff
@@ -500,9 +501,11 @@ func explainResourceOverview(queryResult *util.RunResult, statsInfo *statistic.S
 		if statsInfo != nil {
 			buffer.WriteString("\n")
 			// Calculate the total sum of S3 requests for each stage
-			list, head, put, get, delete, deleteMul := models.CalcTotalS3Requests(gblStats, statsInfo)
-			buffer.WriteString(fmt.Sprintf("\tS3List:%d, S3Head:%d, S3Put:%d, S3Get:%d, S3Delete:%d, S3DeleteMul:%d\n",
-				list, head, put, get, delete, deleteMul,
+			list, head, put, get, delete, deleteMul, writtenRows := models.CalcTotalS3Requests(gblStats, statsInfo)
+
+			s3InputEstByRows := statistic.EstimateS3Input(writtenRows)
```
Comment (xzxiong) [inline pkg/sql/compile/analyze_module.go:506]: 建议，这个函数可以考虑尽量收敛到一个 `models.CalcTotalS3Requests`。按预期这个函数与 TN的行为相关，应减少对对 `statistic.EstimateS3Input` 的感知。

**Review by m-schen — APPROVED** [review]

File: pkg/sql/compile/analyze_module.go (hunk 3 above)
Comment (qingxinhome) [reply pkg/sql/compile/analyze_module.go:506]: OK，i got it

File: pkg/util/trace/impl/motrace/statistic/stats_array.go (hunk 2 above)
Comment (xzxiong) [reply pkg/util/trace/impl/motrace/statistic/stats_array.go]: checked.

File: pkg/util/trace/impl/motrace/statistic/stats_array.go (hunk 1 above)
Comment (xzxiong) [reply pkg/util/trace/impl/motrace/statistic/stats_array.go]: checked.

**Review by XuPeng-SH — APPROVED** [review]

**Review by LeftHandCold — APPROVED** [review]

**Review by triump2020 — APPROVED** [review]

**Pull request merged**